From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 7 Sep 2019 03:36:19 +0900
Subject: ndisc: make first solicit delayed randomly

Closes #13460.

(cherry picked from commit 9973e6c43abb9d3d73fcfee83c886dbd32e700ca)
(cherry picked from commit ccf797511e0af237d67a6a5511e760bd41de1732)
---
 src/libsystemd-network/sd-ndisc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsystemd-network/sd-ndisc.c b/src/libsystemd-network/sd-ndisc.c
index 5e71171..9922203 100644
--- a/src/libsystemd-network/sd-ndisc.c
+++ b/src/libsystemd-network/sd-ndisc.c
@@ -365,7 +365,7 @@ _public_ int sd_ndisc_start(sd_ndisc *nd) {
 
         r = event_reset_time(nd->event, &nd->timeout_event_source,
                              clock_boottime_or_monotonic(),
-                             0, 0,
+                             time_now + USEC_PER_SEC / 2, 1 * USEC_PER_SEC, /* See RFC 8415 sec. 18.2.1 */
                              ndisc_timeout, nd,
                              nd->event_priority, "ndisc-timeout", true);
         if (r < 0)
